### ISSUE TITLE
Feature Request: Add MKL-DNN to speed up dynet conv2d on CPU

### ISSUE THREAD

**Issue by ChijunZhang** [clab/dynet#1337]
Dynet is a great deep learning framework. However, it's conv2d implementation is very slow on CPU, about 6 times slower than PyTorch on a 6700K-CPU machine. Recently Intel released the [MKL-DNN](https://github.com/intel/mkl-dnn) library which can significantly improve Conv2d, Relu, BatchNorm etc performance on Intel CPU platform. For those of us who don`t have GPU, it will be a good News to have it in Dynet.

**Comment by ChijunZhang** [comment]
Glad to hear that :)

**Comment by neubig** [comment]
No. I think we can ask people to install MKL-DNN if they want good performance convolutions on CPU (just like we ask them to do CuDNN for convolutions on GPU).

**Comment by xunzhang** [comment]
Make sense to me.
